src/components/Tag/Tag.stories.mdx

```diff
@@ -0,0 +1,223 @@
+import Tag from './Tag';
+import Typography from '../Typography';
+import Stack from '../storyUtils/Stack';
+import { FIGMA_URL, Function } from '../../utils/common';
+import SectionHeader from '../../storybook/SectionHeader';
+import TagShowcase from '../../storybook/Showcases/TagShowcase';
+import { getIconSelectorKnob } from '../../utils/stories';
+import { useState } from 'react';
+
+<Meta
+  title="Design System/Tag"
+  component={Tag}
+  parameters={{
+    design: [
+      {
+        type: 'figma',
+        name: 'Tag',
+        url: `${FIGMA_URL}?node-id=3325%3A58246`,
+      },
+    ],
+    chromatic: { diffThreshold: 0.3 },
+  }}
+/>
+
+<SectionHeader title={'Tag'} />
+
+- [Overview](#overview)
+- [Props](#props)
+- [Usage](#usage)
+- [Variants](#variants)
+
+## Overview
+
+A universal Tag that is used to highlight system-generated metadata.
+
+## Props
+
+<Props of={Tag} />
+
+## Usage
+
+<UsageGuidelines
+  guidelines={[
+    'Use tags to show metadata that needs to be highlighted somehow',
+    'Use tags to relate secondary information about a page element (px a “Writer” tag next to a name)s',
+    'Use selectable tags to quickly filter existing content',
+    'Do not use tags as standalone elements in a page; they should always relate to something elseh',
+    'Do not use tags for complex filtering. If you want a tag that shows a select menu upon being clicked, used filters instead',
+    'Try not to use selectable tags within tables, as they might confuse the user. Opt for read-only tag variants when possible',
+  ]}
+/>
+
+## Variants
+
+### Read-only Tag
+
+Read-only Tag comes in 5 colors: `neutral`, `blue`, `red`, `purple` and `teal`.
+
+<Preview>
+  <Story name="Read-only Tag">
+    <Stack>
+      <Tag>Label</Tag>
+      <Tag color="blue">Label</Tag>
+      <Tag color="red">Label</Tag>
+      <Tag color="purple">Label</Tag>
+      <Tag color="teal">Label</Tag>
+    </Stack>
+  </Story>
+</Preview>
+
+### Read-only Tag with icon
+
+An leading icon can be added to the read-only Tag.
+
+<Preview>
+  <Story name="Read-only Tag with icon">
+    <Stack>
+      <Tag iconName={'dashboard'}>Label</Tag>
+      <Tag iconName={'dashboard'} color="blue">
+        Label
+      </Tag>
+      <Tag iconName={'dashboard'} color="red">
+        Label
+      </Tag>
+      <Tag iconName={'dashboard'} color="purple">
+        Label
+      </Tag>
+      <Tag iconName={'dashboard'} color="teal">
+        Label
+      </Tag>
+    </Stack>
+  </Story>
+</Preview>
+
+### Selectable Tag
+
+Tag can also be selectable. To enable this functionality, provide the `onSelect` prop to the component.<br/>
+The `isSelected` prop indicates whether the Tag is selected or not.
+
+<Preview>
+  <Story name="Selectable Tag">
+    <Stack>
+      <Function>
+        {() => {
+          const [isSelected, setIsSelected] = useState(false);
+          return (
+            <Tag isSelected={isSelected} onSelect={() => setIsSelected(true)}>
```

Comment:
I would change this to toggle, I would also provide the selected part from the initial state for both user and chromatic to capture

```suggestion
            <Tag isSelected={isSelected} onSelect={() => setIsSelected((isSelected) => !isSelected)}>
```